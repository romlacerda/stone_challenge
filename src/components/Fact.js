import React, { Component } from 'react'
import { View, Text } from 'react-native'
import style from '../style/main'

export default props => 
    <View style={style.fact_box}>
        <Text style={style.fontLess80}>{props.value}</Text>
    </View>