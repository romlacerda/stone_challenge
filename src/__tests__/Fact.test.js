import React from 'react';
import renderer from 'react-test-renderer';
import Fact from '../components/Fact';

// FACTS (II) - Nem todos os fatos podem ter uma categoria
//associada. Nesse caso, um label UNCATEGORIZED
//deve ser associado à entrada na lista. Essa lógica de
//front-end deve ser testável.

// FACTS (II) - As entradas da lista podem variar muito com o
// tamanho do fato em si : para isso, você deve adotar
// dois tamanhos de fonte diferentes de acordo com o
// número de caracteres do Chuck Norris Fact. Use 80
// caracteres como critério de decisão. Essa lógica de
// front-end deve ser testável.

describe('Checar CSS da quantidade de caracteres', () => {
  test('snapshot renders', () => {
    const categories = ['DOG'];
    const valor = 'in brightest day in blackest night no evil shall escape my sight let those who worship evils might, beware my power, green lanterns light!';
    const valorMenor = 'Teste'

    const component = renderer.create(<Fact value={valor} categories={categories}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

