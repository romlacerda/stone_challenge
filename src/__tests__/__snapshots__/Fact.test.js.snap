// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checar CSS da quantidade de caracteres snapshot renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ecf3f3",
      "borderColor": "#14aa4b",
      "borderWidth": 1,
      "margin": 15,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    in brightest day in blackest night no evil shall escape my sight let those who worship evils might, beware my power, green lanterns light!
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 0,
          "margin": 5,
          "marginTop": 20,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        DOG
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 30,
          },
          Object {
            "color": "gray",
            "marginTop": 20,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      Ó†ç
    </Text>
  </View>
</View>
`;
